packages/runtime/src/widgets/__tests__/Button.test.tsx

```diff
@@ -228,4 +230,198 @@ test("test button loading using bindings", async () => {
+            type="button"
+          >
+            Update Label
+          </button>
+          <button
+            data-testid="styles-updater"
+            onClick={() => setStyles({ color: "blue" })}
+            type="button"
+          >
+            Update Styles
+          </button>
+        </div>
+      );
+    };
+
+    const { getByTestId } = render(<Parent />);
+
+    // Initial render
+    expect(renderCount.current).toBe(1);
+
+    // Update label
+    // eslint-disable-next-line @typescript-eslint/require-await
+    await act(async () => {
+      fireEvent.click(getByTestId("label-updater"));
+    });
+
+    // Should re-render once because prop changed
+    expect(renderCount.current).toBe(2);
+
+    // Update styles
+    // eslint-disable-next-line @typescript-eslint/require-await
+    await act(async () => {
+      fireEvent.click(getByTestId("styles-updater"));
+    });
+
+    // Should re-render once because prop changed
+    expect(renderCount.current).toBe(3);
+  });
+
+  test("should handle debounced updates via useRegisterBindings", async () => {
+    const { TestButton, renderCount } = createButtonWithRenderCounter();
+    const MemoizedTestButton = React.memo(TestButton);
+
+    const Parent = () => {
+      const [label, setLabel] = useState("Initial Label");
+
+      return (
+        <div>
+          <MemoizedTestButton
+            id="test-button"
+            label={label}
+            options={{ debounceMs: 100 }}
+          />
+          <button
+            data-testid="rapid-updater"
+            onClick={() => {
+              setLabel("Label 1");
+              setTimeout(() => setLabel("Label 2"), 50);
+              setTimeout(() => setLabel("Label 3"), 75);
+            }}
+            type="button"
+          >
+            Rapid Update
+          </button>
+        </div>
+      );
+    };
+
+    const { getByTestId } = render(<Parent />);
+    expect(renderCount.current).toBe(1);
+
+    await act(async () => {
+      fireEvent.click(getByTestId("rapid-updater"));
+      await new Promise((resolve) => {
+        setTimeout(resolve, 150);
+      });
+    });
+
+    // Should have debounced multiple updates into fewer renders
+    expect(renderCount.current).toBeLessThan(4);
+  });
+
+  test("should re-render when bindings change", async () => {
+    const { TestButton, renderCount } = createButtonWithRenderCounter();
+    const MemoizedTestButton = React.memo(TestButton);
+
+    const Parent = () => {
+      return (
+        <div>
+          <MemoizedTestButton
+            id="test-button"
+            // eslint-disable-next-line no-template-curly-in-string
+            label="${ensemble.storage.get('btnLabel')}"
+            options={{ debounceMs: 100 }}
+          />
+          <Button
+            label="Change button label"
+            onTap={{
+              executeCode:
+                "ensemble.storage.set('btnLabel', 'Ensemble Button')",
```

Comment:
You need a test that checks the render count when one of the bindings on the actions change.

Reply:
@sagardspeed2 did you add this test somewhere?